eng/pipelines/pr-breaking-check.yml

```diff
@@ -0,0 +1,94 @@
+              $packageJson.devDependencies["@azure-tools/typespec-azure-resource-manager"] = "file:$(Build.SourcesDirectory)/typespec-azure/packages/typespec-azure-resource-manager"
+              $packageJson.devDependencies["@azure-tools/typespec-client-generator-core"] = "file:$(Build.SourcesDirectory)/typespec-azure/packages/typespec-client-generator-core"
+              $packageJson.devDependencies["@azure-tools/typespec-azure-rulesets"] = "file:$(Build.SourcesDirectory)/typespec-azure/packages/typespec-azure-rulesets"
+              $packageJson.devDependencies["@typespec/compiler"] = "file:$(Build.SourcesDirectory)/typespec-azure/core/packages/compiler"
+              $packageJson.devDependencies["@typespec/http"] = "file:$(Build.SourcesDirectory)/typespec-azure/core/packages/http"
+              $packageJson.devDependencies["@typespec/openapi"] = "file:$(Build.SourcesDirectory)/typespec-azure/core/packages/openapi"
+              $packageJson.devDependencies["@typespec/openapi3"] = "file:$(Build.SourcesDirectory)/typespec-azure/core/packages/openapi3"
+              $packageJson.devDependencies["@typespec/rest"] = "file:$(Build.SourcesDirectory)/typespec-azure/core/packages/rest"
+              $packageJson.devDependencies["@typespec/versioning"] = "file:$(Build.SourcesDirectory)/typespec-azure/core/packages/versioning"
+
+              $packageJson | ConvertTo-Json -Depth 100 | Out-File -Path "package.json" -Encoding utf8 -NoNewline -Force
+            displayName: "Update package.json in azure-rest-api-specs"
+            workingDirectory: $(Build.SourcesDirectory)/azure-rest-api-specs
+          - script: npm install
+            displayName: Install updated TypeSpec dependencies
+            workingDirectory: $(Build.SourcesDirectory)/azure-rest-api-specs
+          - pwsh: ./eng/scripts/TypeSpec-Generate-Swagger.ps1 -CheckAll
+            displayName: "Generate all swagger files"
+            workingDirectory: $(Build.SourcesDirectory)/azure-rest-api-specs
+          - pwsh: |
+              git restore package.json
+              git restore package-lock.json
+            displayName: Discard versions change
+            workingDirectory: $(Build.SourcesDirectory)/azure-rest-api-specs
+
+          # Step 3: Create PR
+          - checkout: azure-sdk-tools
+            fetchDepth: 1
+          - template: /eng/common/pipelines/templates/steps/git-push-changes.yml@azure-sdk-tools
+            parameters:
+              BaseRepoBranch: auto-update-swagger-$(Build.BuildNumber)
+              BaseRepoOwner: azure-sdk
+              CommitMsg: Update swagger with TypeSpec built from $(Build.BuildNumber)
+              TargetRepoOwner: Azure
+              TargetRepoName: azure-rest-api-specs
+              WorkingDirectory: $(Build.SourcesDirectory)/azure-rest-api-specs
+              ScriptDirectory: $(Build.SourcesDirectory)/azure-sdk-tools/eng/common/scripts
+          - task: PowerShell@2
```

Comment:
How is creating a PR going to help if everything was reverted here? What is the purpose?

Reply:
What you mean by "everything was reverted here"? I only revert package.json and package-lock.json, which is not what we care about. We care about swagger change. An example is [here](https://github.com/Azure/azure-rest-api-specs/pull/29698).

Reply:
I see, so from experience with the spec repo most of the change we need to apply are typespec changes in the typespec-next branch so wondering if this just does a small portion of the solution

Reply:
also because of hte typespec-next branch I feel like this is just going to fail anyway if you run on main

Reply:
I didn't find the typespec-next branch you are saying. Could you point me to it?

I want to clarify the user story here:
1. Developer does some change to typespec-azure and send out a PR with PR number like `abc`.
2. Developer go to this pipeline UI and type in the target branch is `ref/pulls/abc/head` or `ref/pulls/abc/merge`. Then run. You could see the screenshot in the description.

Therefore, our target is WIP change, instead of the code already merged into main.

Reply:
typespec-next branch https://github.com/Azure/azure-rest-api-specs/tree/typespec-next
See for example the last release the changes involved https://github.com/Azure/azure-rest-api-specs/pull/29440

There is many little things that would not count as breaking change but still need a little update to the tsp/swagger like formatting.

This pipeline would basically only work up to the first change.

Reply:
I see. I understand it is very possible (even maybe 100%) we expect to change tsp before each release. But our intention is to uncover all the unexpected change in swagger, which might come from bug or lack of consideration. Every time you are not confident in your change, you could run the pipeline **before** merge. The granularity is for each RP. We might expect tsp change for each release, but we won't expect tsp change for each PR, right?

Reply:
Ok but if you test against main you will always have diff from previous PRs that might have had expected change as well

Reply:
I feel like we get this protection from the azure-rest-api-specs typespec-next ci, which automatically takes up the developer versions and validates them against typespec-next.

It would be *nice* to have a more automated way to check a particular PR against typespec-next directly

Reply:
Could you point me to typespec-next ci? Actually I want to understand how it works.

Reply:
It is just teh `TypeSpecValidation - All` pipeline from teh spec repo that is run in the typespec-next branch on every commit as well as as soon as one of the typespec and typespec azure pipeline finish https://dev.azure.com/azure-sdk/public/_build?definitionId=6143&_a=summary